R/sjPlotFA.R

```diff
@@ -0,0 +1,381 @@
+#' sjp.fa(likert_4, type = "tile")
+#' 
+#' # plot results from PCA as bars
+#' sjp.fa(likert_4, type = "bar")
+#' 
+#' 
+#' # -------------------------------
+#' # Data from the EUROFAMCARE sample dataset
+#' # -------------------------------
+#' library(sjmisc)
+#' data(efc)
+#' 
+#' # retrieve variable and value labels
+#' varlabs <- get_label(efc)
+#' 
+#' # recveive first item of COPE-index scale
+#' start <- which(colnames(efc) == "c82cop1")
+#' # recveive last item of COPE-index scale
+#' end <- which(colnames(efc) == "c90cop9")
+#'  
+#' # create data frame with COPE-index scale
+#' mydf <- data.frame(efc[, c(start:end)])
+#' colnames(mydf) <- varlabs[c(start:end)]
+#' 
+#' sjp.fa(mydf)
+#' sjp.fa(mydf, type = "tile")
+#' 
+#' # -------------------------------
+#' # auto-detection of labels
+#' # -------------------------------
+#' sjp.fa(efc[, c(start:end)], type = "circle", geom.size = 10)
+#' 
+#' 
+#' @import ggplot2
+#' @importFrom tidyr gather
+#' @importFrom scales brewer_pal grey_pal
+#' @importFrom stats na.omit prcomp
+#' @importFrom sjstats cronb
+#' @importFrom psych fa
+#' @importFrom psych fa.parallel
+#' @export
+sjp.fa <- function(data,
+                    rotation = c("promax", "varimax"),
+                    method = "ml",
+                    nmbr.fctr = NULL,
+                    fctr.load.tlrn = 0.1,
+                    digits = 2,
+                    title = NULL,
+                    axis.labels = NULL,
+                    type = c("bar", "circle", "tile"),
+                    geom.size = .6,
+                    geom.colors = "RdBu",
+                    wrap.title = 50,
+                    wrap.labels = 30,
+                    show.values = TRUE,
+                    show.cronb = TRUE,
+                    prnt.plot = TRUE) {
+  # --------------------------------------------------------
+  # check arguments
+  # --------------------------------------------------------
+  type <- match.arg(type)
+  rotation <- match.arg(rotation)
+  # --------------------------------------------------------
+  # try to automatically set labels is not passed as argument
+  # --------------------------------------------------------
+  if (is.null(axis.labels) && is.data.frame(data)) {
+    axis.labels <- unname(sjmisc::get_label(data, def.value = colnames(data)))
+  }
+  # ----------------------------
+  # set color palette
+  # ----------------------------  
+  if (is.brewer.pal(geom.colors[1])) {
+    geom.colors <- scales::brewer_pal(palette = geom.colors[1])(5)
+  } else if (geom.colors[1] == "gs") {
+    geom.colors <- scales::grey_pal()(5)
+  }
+  # ----------------------------
+  # check if user has passed a data frame
+  # or a pca object
+  # ----------------------------
+  if (inherits(data, "fa")) {
+    fadata <- data
+    dataframeparam <- FALSE
+  } else if (is.data.frame(data)) {
+    
+    if (is.null(nmbr.fctr)) {
+      nr_factors <- psych::fa.parallel(data, fa = 'fa', fm = method)$nfact
+      dev.off()
```

Comment:
ah, I see. I think, inside a function, you just need to do nothing. Rather, you have to explicitely plot the plot - unless there are some quirks in `fa.parallel()`, which gives you no chance to suppress the plot.